Source/NSMovie.m

```diff
@@ -21,88 +23,112 @@
 
    You should have received a copy of the GNU Lesser General Public
    License along with this library; see the file COPYING.LIB.
-   If not, see <http://www.gnu.org/licenses/> or write to the 
-   Free Software Foundation, 51 Franklin Street, Fifth Floor, 
+   If not, see <http://www.gnu.org/licenses/> or write to the
+   Free Software Foundation, 51 Franklin Street, Fifth Floor,
    Boston, MA 02110-1301, USA.
 */
 
 #import <Foundation/NSArray.h>
 #import <Foundation/NSCoder.h>
 #import <Foundation/NSData.h>
 #import <Foundation/NSURL.h>
+#import <Foundation/NSFileManager.h>
+#import <Foundation/NSUUID.h>
+
 #import "AppKit/NSMovie.h"
 #import "AppKit/NSPasteboard.h"
 
+NSString *_writeDataToTempFile(NSData *data)
+{
+  NSString *tempDirectory = NSTemporaryDirectory();
+  NSString *filename = [NSString stringWithFormat: @"tmpfile-%@.dat", [[NSUUID UUID] UUIDString]];
+  NSString *filepath = [tempDirectory stringByAppendingPathComponent: filename];
+  NSError *error = nil;
+
+  BOOL success = [data writeToFile: filepath options: NSDataWritingAtomic error: &error];
+  if (success)
+    {
+      return nil;
+    }
+
+  return filepath;
+}
+
 @implementation NSMovie
 
-+ (NSArray*) movieUnfilteredFileTypes
++ (NSArray *) movieUnfilteredFileTypes
 {
-  return [NSArray arrayWithObject: @"mov"];
+  return [NSArray array];
 }
 
-+ (NSArray*) movieUnfilteredPasteboardTypes
++ (NSArray *) movieUnfilteredPasteboardTypes
 {
-  // FIXME
-  return [NSArray arrayWithObject: @"QuickTimeMovie"];
+  return [NSArray array];
 }
 
-+ (BOOL) canInitWithPasteboard: (NSPasteboard*)pasteboard
++ (BOOL) canInitWithPasteboard: (NSPasteboard *)pasteboard
 {
   NSArray *pbTypes = [pasteboard types];
   NSArray *myTypes = [self movieUnfilteredPasteboardTypes];
 
   return ([pbTypes firstObjectCommonWithArray: myTypes] != nil);
 }
 
-- (id) initWithData: (NSData *)movie
+- (instancetype) initWithData: (NSData *)movie
 {
   if (movie == nil)
     {
       RELEASE(self);
       return nil;
     }
-  
-  [super init];
-  ASSIGN(_movie, movie);
 
-  return self;
-}
+  self = [super init];
+  if (self != nil)
+    {
+      NSString *filepath = _writeDataToTempFile(_movie);
 
-- (id) initWithMovie: (void*)movie
-{
-  //FIXME
+      _url = [NSURL fileURLWithPath: filepath];
+      _tmp = YES;
+      ASSIGN(_movie, movie);
+    }
 
   return self;
 }
 
-- (id) initWithURL: (NSURL*)url byReference: (BOOL)byRef
+- (instancetype) initWithMovie: (void *)movie
 {
-  NSData* data = [url resourceDataUsingCache: YES];
-
-  self = [self initWithData: data];
+  return [self initWithData: movie];
```

Comment:
This seems strange. Shouldn't you  package up the void pointer in an NSData object before passing it on?

Reply:
I wasn't sure how to handle this one.  Given that we really don't know what the pointer being passed in is.  I felt as though it was safer to just treat it like an NSData.